lib/matplotlib/tests/test_datetime.py

```diff
@@ -50,11 +50,35 @@ def test_axvline(self):
         fig, ax = plt.subplots()
         ax.axvline(...)
 
-    @pytest.mark.xfail(reason="Test for axvspan not written yet")
     @mpl.style.context("default")
     def test_axvspan(self):
-        fig, ax = plt.subplots()
-        ax.axvspan(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        np.random.seed(19680801)
+
+        start_date = datetime.datetime(2023, 1, 1)
+        time_delta = datetime.timedelta(days=1)
+
+        values1 = np.random.randint(1, 10, 30)
+        values2 = np.random.randint(1, 10, 30)
+        values3 = np.random.randint(1, 10, 30)
+
+        bin_edges = [start_date + i * time_delta for i in range(31)]
+
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1, constrained_layout=True)
+
+        axes = [ax1, ax2, ax3]
+        values_list = [values1, values2, values3]
+
+        for ax, values in zip(axes, values_list):
```

Comment:
The 3 subplots aren't testing anything different right now. Could you please update the test so that the three subplots are plotting these different combinations on the X and Y axes:

1. Datetime vs number
2. Number vs datetime
3. Datetime vs datetime

Reply:
Yes I will fix it tomorrow.

Reply:
Hello, just made an update. Hope it will be OK.